test/e2e/tests/resize_e2e_test.go

```diff
@@ -0,0 +1,241 @@
+	Expect(err).To(BeNil(), "NodeUnpublishVolume failed with error")
+}
+
+func onlineResizeDisk(di *DiskInfo) {
+	instance := di.TestCtx.Instance
+	volName := di.Name
+
+	// Stage Disk
+	stageDir := filepath.Join("/tmp/", volName, "stage")
+	err := testutils.MkdirAll(instance, stageDir)
+	Expect(err).To(BeNil(), "Mkdir failed with error")
+
+	err = di.TestCtx.Client.NodeStageVolume(di.Volume.GetVolumeId(), stageDir, di.Volume.GetVolumeContext())
+	Expect(err).To(BeNil(), "NodeStageVolume failed with error")
+	defer func() {
+		// Unstage Disk
+		err := di.TestCtx.Client.NodeUnstageVolume(di.Volume.GetVolumeId(), stageDir)
+		Expect(err).To(BeNil(), "NodeUnstageVolume failed with error")
+
+		// delete remote directory
+		fp := filepath.Join("/tmp/", volName)
+		err = testutils.RmAll(instance, fp)
+		Expect(err).To(BeNil(), "Failed to delete remote directory")
+	}()
+
+	// Mount Disk
+	publishDir := filepath.Join("/tmp/", volName, "mount")
+	err = testutils.MkdirAll(instance, publishDir)
+	Expect(err).To(BeNil(), "Mkdir failed with error")
+	err = di.TestCtx.Client.NodePublishVolume(di.Volume.GetVolumeId(), stageDir, publishDir, di.Volume.GetVolumeContext())
+	Expect(err).To(BeNil(), "NodePublishVolume failed with error")
+	defer func() {
+		// Unmount Disk
+		err := di.TestCtx.Client.NodeUnpublishVolume(di.Volume.GetVolumeId(), publishDir)
+		Expect(err).To(BeNil(), "NodeUnpublishVolume failed with error")
+	}()
+
+	// Resize
+
+	// Resize without limit to nearest Gb
+	newSizeBytes := minVolumeSize + (10 * util.Gb) + 1
+	err = di.TestCtx.Client.ControllerExpandVolume(di.Volume.GetVolumeId(), newSizeBytes)
+	Expect(err).To(BeNil(), "Controller expand volume failed for resize without limit to nearest Gb")
+	validateResizeDisk(newSizeBytes, di, "online resize - resize to nearest Gb")
+
+	// Resize with limit too small
+	oldSizeBytes := newSizeBytes
+	newSizeBytes = minVolumeSize * 3
+	newSizeLimit := minVolumeSize * 2
+	err = di.TestCtx.Client.ControllerExpandVolumeWithLimit(di.Volume.GetVolumeId(), newSizeBytes, newSizeLimit)
+	Expect(err).ToNot(BeNil(), "Controller expand volume unexpected success for resize with invalid limit")
+	validateResizeDisk(oldSizeBytes, di, "online resize - resize with invalid limit")
+
+	// Resize with limit
+	newSizeBytes = minVolumeSize * 2
+	newSizeLimit = minVolumeSize * 3
+	err = di.TestCtx.Client.ControllerExpandVolumeWithLimit(di.Volume.GetVolumeId(), newSizeBytes, newSizeLimit)
+	Expect(err).To(BeNil(), "Controller expand volume failed for resize with valid limit")
+	validateResizeDisk(newSizeBytes, di, "online resize - resize with valid limit")
+
+	// Invalid resize to smaller amount
+	err = di.TestCtx.Client.ControllerExpandVolume(di.Volume.GetVolumeId(), minVolumeSize)
+	Expect(err).To(BeNil(), "Controller expand volume should not error for invalid resize to smaller amount.")
+	validateResizeDisk(newSizeBytes, di, "online resize - resize to smaller amount")
+}
```

Comment:
Do we want to test any writing to the disk to make sure it wasn't unmounted / reformatted during the resize?

Reply:
+1 for offline resize test too. after resize, then stage/publish and test write

Reply:
Done - refactored the write / read as well into their own respective methods for easier testing.